src/ui/digital-exchange/components/list/ComponentList.js

```diff
@@ -1,27 +1,48 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import { Spinner, Alert } from 'patternfly-react';
+import { Spinner, Alert, Paginator } from 'patternfly-react';
 import { FormattedMessage } from 'react-intl';
 
 import ComponentListGridView from 'ui/digital-exchange/components/list/ComponentListGridView';
 import ComponentListListView from 'ui/digital-exchange/components/list/ComponentListListView';
+import ExtraTabBarFilterContainer from 'ui/digital-exchange/ExtraTabBarFilterContainer';
 
 import { DE_COMPONENTS_GRID_VIEW } from 'state/digital-exchange/components/const';
 import { componentType } from 'models/digital-exchange/components';
 
 
 class ComponentList extends Component {
+  constructor(props) {
+    super(props);
+    this.changePage = this.changePage.bind(this);
+    this.changePageSize = this.changePageSize.bind(this);
+  }
+
   componentWillMount() {
     this.props.onWillMount();
   }
 
+  changePage(page) {
+    this.props.fetchDEComponentsFiltered({ page, pageSize: this.props.pageSize });
+  }
+
+  changePageSize(pageSize) {
+    this.props.fetchDEComponentsFiltered({ page: 1, pageSize });
+  }
+
   render() {
     const {
       loading,
       viewMode,
       digitalExchangeComponents,
     } = this.props;
 
+    const pagination = {
+      page: this.props.page,
+      perPage: this.props.pageSize,
```

Comment:
de-struct `page` and `perPage` above and just use the var as key in `pagination`
```suggestion
      loading,
      viewMode,
      digitalExchangeComponents,
      page,
      perPage,
    } = this.props;

    const pagination = {
      page,
      perPage,
```